### PR TITLE
Adds relative path storage option.

### PR DESCRIPTION
**Pull request by kovasap**

### CODE REVIEW

**Comment by pbogut** [comment]
I have mixed feelings about it. Do you have any weird behaviour when using it? How it acts if you open MRU in different projects? Isn't it creating new buffers when you try to open recent file in different directory? Or removing files from the list?

Since it's small enough change it doesn't look like it will break anything for ppl who would not use that option, but I'm just curious.

**Comment by kovasap** [comment]
Let me use it for a while longer to find/fix any bugs (or realize there are some fundamental problems :)).

**Comment by kovasap** [comment]
I spent a bit more time with this and realized that it's actually more useful if the option is a directory pattern, so that the relative file finding is only done when in the repositories that have multiple copies at different absolute paths.  This cleans up any issues I was having with it like you mentioned - e.g. multiple versions of a file appearing if I opened them from different dirs.  Using this properly requires you are always at the root of the directory that matches the pattern.

**Comment by pbogut** [comment]
What kind of issues we are talking about exactly? And how you imagine this option to look like?

I'm thinking, we could add filter for recent files and when using relative files always check if file exists. On upside it would show only files that are actually available and are relative to opened directory. Downside would probably be performance hit (although it could only check at the start) and possible it would show some files that are common in many projects, like README.md etc.
Thoughts?

**Comment by kovasap** [comment]
The main issue I ran into was that I would accidentally open files in another version of my repository.  Here is an example:

```
/path/to/repo/version_1/path/to/file.py
/path/to/repo/version_2/path/to/file.py
/path/to/repo/version_3/path/to/file.py
```

If my CWD is `/path/to/repo/version_1/`, i want to only see/open `path/to/file.py`.  I was having issues where the other files with absolute paths were showing up, and I would inadvertently start editing them, when I should never be interacting with files in a version directory that I am not within.  This was the initial inspiration for this option.

I want to make sure that if a new version that I haven't interacted with (e.g. `version_4`) appears, I still get the MRU files from the other repo versions.  

One potential way to do this is to capture absolute paths always, but prune off the front of the path if the directory matches the pattern (that should match the version_N directory as well).  But the current code works for me.  

I also have normal git repos (and other files) that I also want to interact with with absolute paths, which is why I made the change to only look for relative files if the directory pattern matches (`/path/to/repo/` in this case).

**Comment by kovasap** [comment]
Does this make sense to you and/or did I address your concerns?  If so, I can change the documentation part of this PR before it's merged.

**Comment by kovasap** [comment]
Following up again on this.  What do you think of the change in it's current state?

**Comment by pbogut** [comment]
@kovasap To be honest I don't like it can only be set for `one` project. I mean, I guess its good enough for you, but if something like that has to be there I would rather have an option to set up multiple project directories.

It would be good if that option was list instead of string.

```
let g:store_relative_dirs = ['/path/to/repo/', '/path/to/different/repo/']
```

**Comment by kovasap** [comment]
Good idea!  Added that option.

File: autoload/fzf_mru/mrufiles.vim [hunk 1]
```diff
@@ -68,6 +69,13 @@ endf
 
 fu! s:addtomrufs(fname)
   let fn = fnamemodify(a:fname, ':p')
+  for rdir in {s:stre}
+    if getcwd() =~ rdir
+      let fn = fnamemodify(a:fname, ':.')
+      break
+    endif
+  endfor
+  " if !empty({s:stre}) && getcwd() =~ {s:stre}
```
Comment (pbogut) [inline autoload/fzf_mru/mrufiles.vim]: I see some leftovers.

**Review by pbogut — COMMENTED** [review]
You left some code commented, can you please clean it up?. I'm happy with the rest of the PR, can merge when its cleared.

**Comment by kovasap** [comment]
Oh, my bad, fixed!

**Comment by pbogut** [comment]
Thank you for your contribution.

**Comment by kovasap** [comment]
You're welcome!  Thanks for accepting.

**Comment by pbogut** [comment]
Not sure if you care or not, but you are using different email in git configuration than in Github. Because of that your commits are not connected to your Github account, and you are not listed in contributors etc. I think you can fix that by adding your git email (you can check it here `git config --global user.email`) to Github emails (you can add multiple emails). If you don't care just ignore me.

**Comment by kovasap** [comment]
Oh thanks for noticing!  I updated my emails accordingly and it looks like on my commits my github name is now used (before it was just my full name).  I don't see my name in the contributors section yet, maybe i'll give it some time to propogate?

**Comment by pbogut** [comment]
Maybe it needs some time. When I refresh commit page it's sometimes github name sometimes full name, so it looks like it still didn't propagate everywhere.

**Comment by kovasap** [comment]
Hmm, still not working, I'm OK with just ignoring it :).

**Pull request merged**